models/src/anemoi/models/layers/block.py

```
@@ -63,6 +63,35 @@ def forward(
     ) -> tuple[torch.Tensor, torch.Tensor]: ...
 
 
+class PointWiseMLPProcessorBlock(BaseBlock):
+    """Point-wise block with MLPs."""
+
+    def __init__(self, *, num_channels: int, hidden_dim: int, layer_kernels: DotDict, dropout_p: float = 0.0):
+        super().__init__()
+        layers = [
+            layer_kernels.Linear(num_channels, hidden_dim),
+            layer_kernels.LayerNorm(hidden_dim),
+            layer_kernels.Activation(),
+        ]
+        if num_channels != hidden_dim:
+            layers.append(layer_kernels.Linear(hidden_dim, num_channels))
+
+        if dropout_p is not None and dropout_p > 0:
```

It might better to raise an error for `dropout <0` instead of letting it fall through.

---

Thanks, check added now! I think the purpose of setting `dropout_p > 0` here was to avoid instantiating a Dropout(0).